src/Tracing/SpanContext.php

```diff
@@ -19,16 +19,10 @@ class SpanContext
 
     public static function generate()
     {
-        return self::fork(bin2hex(random_bytes(16)));
+        return bin2hex(random_bytes(16));
     }
 
-    // This method is probably useful, but it's name is confusing
-    public static function fork(string $traceId)
-    {
-        return self::restore($traceId, bin2hex(random_bytes(8)));
-    }
-
-    public function __construct(string $traceId, string $spanId, string $traceFlags, array $traceState)
+    public function __construct(string $traceId, string $spanId, ?string $traceFlags = null, ?array $traceState = null)
```

Comment:
I made $traceFlags and $traceState optional here so that we could appease the $context creation in src/Tracing/Tracer.php